DuckDuckGo/TabViewController.swift

```diff
@@ -1000,7 +1000,14 @@ extension TabViewController: WKNavigationDelegate {
         
         isShowingFullScreenDaxDialog = true
         scheduleTrackerNetworksAnimation(collapsing: !spec.highlightAddressBar)
+        let daxDialogSourceURL = self.url
+        
         DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
+            // https://app.asana.com/0/414709148257752/1201620790053163/f
+            if self?.url != daxDialogSourceURL {
+                return
+            }
```

Comment:
Question: won't that simply omit the Dax dialog, so given message just disappears and is never shown again?

Reply:
You're right, but the main problem is that DaxDialogs is deciding if itself was shown or not, for example:
https://github.com/duckduckgo/iOS/blob/123a583a08a313e0ef680824e036a1faaef1fc58/DuckDuckGo/DaxDialogs.swift#L296

This concept is wrong because DaxDialogs doesn't know if it was shown or not, it only creates the spec. Either `TabViewController` or `FullscreenDaxDialogViewController` knows that info, but not the DaxDialog since they're only holding data, not displaying it.


I can think about 2 different solutions that doesn't require a bigger refactor:
### 1 - Only update the flag once the view controller has shown the items.

On DaxDialogs remove all the shown flags = true on the methods where they are created, then add the following method:

```
    func updateShownFlag(_ spec: BrowsingSpec) {
        switch spec.type {
        case .withMutipleTrackers, .withOneTracker :
            settings.browsingWithTrackersShown = true
        case .afterSearch:
            settings.browsingAfterSearchShown = true
        case .withoutTrackers:
            settings.browsingWithoutTrackersShown = true
        case .siteIsMajorTracker, .siteOwnedByMajorTracker:
            settings.browsingMajorTrackingSiteShown = true
            settings.browsingWithoutTrackersShown = true
         }
    }

```
    
which would require a new type to be added since currently the same struct "type" can have different messages so we can't rely on the Equatable for the actual "type"


```
        enum BrowsingSpecType  {
        case afterSearch
        case withoutTrackers
        case siteIsMajorTracker
        case siteOwnedByMajorTracker
        case withOneTracker
        case withMutipleTrackers
    }

```
    
 and then on `FullScreenDaxDialogViewController`  can call:     `DaxDialogs.shared.updateShownFlag(spec)`

This approach makes sure the flag is only updated at the correct time, BUT, it breaks all the tests (which are quite a lot), and it's more disruptisvde since it's a paradigm shift.


### 2 - Add a method that overrides the flag

Similar to approach 1, we'd have a method on DaxDialogs:
```
    func overrideShownFlagForSpec(spec: BrowsingSpec, flag: Bool) {
        switch spec.type {
        case .withMutipleTrackers, .withOneTracker :
            settings.browsingWithTrackersShown = flag
        case .afterSearch:
            settings.browsingAfterSearchShown = flag
        case .withoutTrackers:
            settings.browsingWithoutTrackersShown = flag
        case .siteIsMajorTracker, .siteOwnedByMajorTracker:
            settings.browsingMajorTrackingSiteShown = flag
            settings.browsingWithoutTrackersShown = flag
         }
    }

```

But we'd leave the assigments to the flags on the methods that creates them like we have today.
And on TabViewController we add:
```
            if self?.url != daxDialogSourceURL {
                DaxDialogs.shared.overrideShownFlagForSpec(spec: spec, flag: false)
                return
            }


```
Which would cause the flag to be removed.

I'll go with solution number 2 because It doesn't change the paradigm of how the flags are set and doesn't break the tests. 

Please let me know if you disagree and/or have a suggestion for this.
Thank you!